src/libbson/tests/test-json.c

```diff
@@ -2841,6 +2841,701 @@ test_bson_as_json_multi_object (void)
    TEST_JSON_PRODUCES_MULTIPLE ("[],[{'a': 1}]", 1, NULL);
 }
 
+#define TEST_BSON_AS_JSON_WITH_OPTS(_bson, _mode, _max_len, _expected) \
+   {                                                                   \
+      bson_json_opts_t _opts = {_mode, _max_len};                      \
+      size_t _json_len;                                                \
+      char *_str = bson_as_json_with_opts (_bson, &_json_len, &_opts); \
+      ASSERT_CMPSTR (_str, _expected);                                 \
```

Comment:
Suggest adding another check to ensure that the produced string does not exceed maximum specified length when truncation occurs.
```
if (_max_len != BSON_MAX_LEN_UNLIMITED) {
    ASSERT_CMPINT(_json_len, <=, _max_len);
}
```